pkg/storage/gcp/gcp_test.go

```diff
@@ -2,14 +2,27 @@ package gcp
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"testing"
 
 	"github.com/gomods/athens/pkg/config"
 	"github.com/gomods/athens/pkg/storage/compliance"
+	"github.com/technosophos/moniker"
 	"google.golang.org/api/iterator"
 )
 
+var bucketName = randomBucketName(os.Getenv("DRONE_PULL_REQUEST"))
+var strg *Storage
+
+func TestMain(m *testing.M) {
+	exitCode := m.Run()
+	if strg != nil {
+		strg.bucket.Delete(context.Background())
+	}
```

Comment:
@marwan-at-work would you be able to create and delete a bucket inside of each the individual test and benchmark? I think that would be a little easier to follow than the globals.

Reply:
@arschles that would require a refactoring of all the tests, we can do that as a follow up in my opinion if we deem it necessary. It would also be more costly because you'd be creating/destroying buckets 6x more per PR ;)

Reply:
Oh, I didn't see all 6 places. Where are the others beside `TestBackend` and `BenchmarkBackend`?

Reply:
@arschles to create/remove bucket for every single test suite, you'd have to have a new "setup" function before every test suite in this file: https://github.com/gomods/athens/blob/master/pkg/storage/compliance/tests.go

Which means every single storage we have tests for should update its signatures. 

It's a totally good solution and would allow our tests to be more flexible but it's a big lift for just fixing the ci/cd part :)

Reply:
@marwan-at-work I think I'm missing something because you already call `getStorage` at the beginning of `TestBackend` and `BenchmarkBackend` in the GCP tests. It seems like the only difference is that you need the global `strg` variable so that `TestMain` works properly, but `TestMain` doesn't need to exist in these GCP tests, does it?

Reply:
@arschles you're so right! I forgot that we actually call `compliance.RunTests` from with the test function itself, so we can actually defer cleaning up. 

Thanks for the call out :) I updated the code

Reply:
feel free to comment more, cuz i mighta missed a couple things here and there